src/mono/browser/runtime/types/internal.ts

```diff
@@ -569,36 +567,37 @@ export interface MonoThreadMessage {
     cmd: string;
 }
 
-// keep in sync with JSHostImplementation.Types.cs
-export const enum MainThreadingMode {
-    // Running the managed main thread on UI thread.
-    // Managed GC and similar scenarios could be blocking the UI.
-    // Easy to deadlock. Not recommended for production.
-    UIThread = 0,
-    // Running the managed main thread on dedicated WebWorker. Marshaling all JavaScript calls to and from the main thread.
-    DeputyThread = 1,
-    // TODO comment
-    DeputyAndIOThreads = 2,
-}
-
 // keep in sync with JSHostImplementation.Types.cs
 export const enum JSThreadBlockingMode {
-    // throw PlatformNotSupportedException if blocking .Wait is called on threads with JS interop, like JSWebWorker and Main thread.
-    // Avoids deadlocks (typically with pending JS promises on the same thread) by throwing exceptions.
-    NoBlockingWait = 0,
-    // TODO comment
-    AllowBlockingWaitInAsyncCode = 1,
-    // allow .Wait on all threads.
-    // Could cause deadlocks with blocking .Wait on a pending JS Task/Promise on the same thread or similar Task/Promise chain.
-    AllowBlockingWait = 100,
-}
-
-// keep in sync with JSHostImplementation.Types.cs
-export const enum JSThreadInteropMode {
-    // throw PlatformNotSupportedException if synchronous JSImport/JSExport is called on threads with JS interop, like JSWebWorker and Main thread.
-    // calling synchronous JSImport on thread pool or new threads is allowed.
-    NoSyncJSInterop = 0,
-    // allow non-re-entrant synchronous blocking calls to and from JS on JSWebWorker on threads with JS interop, like JSWebWorker and Main thread.
-    // calling synchronous JSImport on thread pool or new threads is allowed.
-    SimpleSynchronousJSInterop = 1,
+    /**
+     * Prevents synchronous JSExport from being called from JavaScript code in UI thread.
+     * On JSWebWorker synchronous JSExport always works.
+     * On JSWebWorker blocking .Wait always warns.
+     * This is the default mode.
+     */
+    PreventSynchronousJSExport = 0,
```

Comment:
We should use string values instead of numbers in public API, same as for `GlobalizationMode`

Reply:
We need it as number during early start and on managed side. For config, I agree, good point.